src/compiler/scala/tools/nsc/typechecker/Namers.scala

```diff
@@ -1592,11 +1592,7 @@ trait Namers extends MethodSynthesis {
       import SymValidateErrors._
       def fail(kind: SymValidateErrors.Value) = SymbolValidationError(sym, kind)
 
-      def checkWithDeferred(flag: Int) {
-        if (sym hasFlag flag)
-          AbstractMemberWithModiferError(sym, flag)
-      }
-      def checkNoConflict(flag1: Int, flag2: Int) {
+      def checkNoConflict(flag1: Int, flag2: Int) = {
```

Comment:
That option only considers type arguments inferred as `Any` / `AnyVal`, not the types of `if` / `match` / `try` expressions.